### ISSUE TITLE
Java and jdbc upgrades

### ISSUE THREAD

**Issue by denislepage** [MotusDev/Motus-TO-DO#866]
Two things: 

1) I am planning to upgrade the Java JDBC driver to the latest version.

Based on this, version 8.4 supports the JRE 1.8, 11 and 14.

https://docs.microsoft.com/en-us/sql/connect/jdbc/microsoft-jdbc-driver-for-sql-server-support-matrix?view=sql-server-ver15#java-and-jdbc-specification-support

2) I would also like to move towards updating our Java version relatively soon.

The open JDK versions are what we'll need given Oracle's changes in licensing model. Based on the above, We'll want to make sure if there are compatibility issue of the latest release (version 15), or if we should look at another version.

https://openjdk.java.net/

I am not yet sure if multiple versions can co-exist on our server. There's at least 1 file being shared across all Tomcat instances that is version dependent, but I may be able to localize it. We can give that a try on the Motus sandbox first. I may try to restore the sandbox2 instance initially, which is running a new Tomcat version also.

It's potentially fraught with complications, but we need to bite this bullet.

**Comment by denislepage** [comment]
OK, it looks like the sql_auth file can be localized for each tomcat instance. I have the newest JDBC 8.4 now loaded on the bsc sandbox and working fine. @Hussell I think that once you modify your path to use jdbc 8.4 and compile a new jar, we'll also have to make those changes on the Tomcat instance. I am getting run-time warnings that we will need to ignore for now, but you may not see them during the build. Both jdbc jar files are in the shared-lib folder in bscbase. I haven't yet copied the new one to motus shared-lib.

**Comment by denislepage** [comment]
Let's tackle this jdbc thing first, because I have dependencies in my code (yet in git bscbase master) that expects a newer jdbc, but we'll need to change the jdbc version again when we finally upgrade Java as well.

**Comment by denislepage** [comment]
I bit the JDK bullet.

I have started experimenting with Java JDK 17. So far, I haven't run into too many issues. There were some deprecated constructors for primitive data type (Integer, Double, Long, etc.), so I changed a larger number of files (more so in bscbase). The code should still compile fine on our older Java 8, but the classes compiled with Java 17 won't work on Tomcat running an older version of Java. These changes are already in sandbox and bscbase, but I expect there won't be compatibility issue caused by the commits so far,

Motus servers are running tomcat 8.0, which do not support JDK 17, so Tomcat servers will also need to be updated. 

This is where the biggest change is required so far. Tomcat 10 relies on the Jakarta version of servlets (jakarta.servlet.* instead of javax.servlet.*). I have created new local branch on my computer here that switch from javax to jakarta. At the same time, I have updated a bunch of older jar files.  The apache-fileupload package is no longer compatible, and the same functionalities are now part of the servlet specifications. This affects 3 or 4 classes that I will need to rewrite that handle file uploads.

Once I am done rewriting those cases, I'll do a bit of testing on the newly installed Tomcat 10. If that seems to work, I will look to merge my jakarta branch into master, or at the very least push my jakarta branch upstream.
